File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
import java.util.*;

class Solution {
    static Set<Integer> set = new HashSet<>();
    static PriorityQueue<Integer> pq = new PriorityQueue<>();
    static int length;
    
    public int[] solution(int[] numbers) {
        length = numbers.length;
        
        for(int i=0; i<length; i++) {
            for(int j=i+1; j<length; j++) {
                set.add(numbers[i] + numbers[j]);
            }
        }
        
        pq.addAll(set);
        
        int size = pq.size();
        
        int[] answer = new int[size];
        for(int i=0; i<size; i++)
            answer[i] = pq.poll();
        
        return answer;
    }
}

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;

vector<int> solution(vector<int> numbers) {
    
    int length = numbers.size();
    // 인덱스를 N*N으로 돌면서 각 인덱스의 요소를 더한다.
    // 더한 값을 set에 담는다.
    // 이걸 vector로 만들어서 정렬해 리턴한다.
    
    set<int> s;
    for(int i=0; i<length; i++) {
        for(int j=i+1; j<length; j++) {
            s.insert(numbers[i] + numbers[j]);
        }
    }
    
    vector<int> answer;
    for(int i: s)
        answer.push_back(i);
    
    return answer;
}